src/App.tsx

```diff
@@ -35,61 +35,122 @@ export function getReorderedGoods(
   const visibleGoods = [...goods];
 
   // Sort and reverse goods if needed
+  visibleGoods.sort((a, b) => {
+    switch (sortType) {
+      case SortType.ALPHABET: {
+        return a.localeCompare(b);
+      }
+
+      case SortType.LENGTH: {
+        return a.length - b.length;
+      }
+
+      case SortType.NONE: {
+        return 0;
+      }
+
+      default: {
+        throw new Error('Wrong sorting type!');
+      }
+    }
+  });
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
 
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
   return visibleGoods;
 }
 
 // DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  sortByAlphabet = () => {
+    this.setState(({ sortType: SortType.ALPHABET }));
```

Comment:
why so many brackets? 😄 
```suggestion
    this.setState({ sortType: SortType.ALPHABET });
```